### PR TITLE
Move pending tx handling to Redux

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
This still needs to be tested to make sure it works as intended when moved to a saga.

**Comment by FrederikBolding** [comment]
[ch8342]

File: src/services/Store/store/account.slice.ts [hunk 1]
```diff
@@ -234,3 +255,64 @@ export function* addTxToAccountWorker({
...
+        t.hash !== pendingTxReceipt.hash &&
+        t.status === ITxStatus.SUCCESS
+    );
+
+    if (overwritingTx) {
+      const updatedAccount = {
+        ...senderAccount,
+        transactions: senderAccount.transactions.filter((t) => t.hash !== pendingTxReceipt.hash)
+      };
+      yield put(updateAccount(sanitizeAccount(updatedAccount)));
```
Comment (GuillaumeRx) [inline src/services/Store/store/account.slice.ts:288]: you don't have to call `sanitizeAccount` here, it's done in the reducer :)

File: src/services/Store/store/account.slice.ts [hunk 2]
```diff
@@ -234,3 +255,64 @@ export function* addTxToAccountWorker({
...
+    // txStatus and txTimestamp return undefined on failed lookups.
+    if (!txStatus || !txTimestamp) continue;
+
+    const finishedTxReceipt = makeFinishedTxReceipt(
+      pendingTxReceipt,
+      txStatus,
+      txTimestamp,
+      txResponse.blockNumber
+    );
+    yield put(addTxToAccount({ account: sanitizeAccount(senderAccount), tx: finishedTxReceipt }));
```
Comment (GuillaumeRx) [inline src/services/Store/store/account.slice.ts:316]: same here

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

File: src/services/Store/store/account.slice.ts [hunk 3]
```diff
@@ -148,20 +148,10 @@ export const getAccountsAssetsMappings = createSelector([getAccountsAssets], (as
...
-    );
-    return {
-      ...a,
-      assets: accountAssets,
-      network: getNetwork(a.networkId)(s),
-      label: selectAccountContact(a)(s)?.label || a.label || translateRaw('NO_LABEL')
-    };
+    const contact = selectAccountContact(a)(s);
+    const network = getNetwork(a.networkId)(s);
+    const assets = getAssets(s);
```
Comment (GuillaumeRx) [inline src/services/Store/store/account.slice.ts:153]: can we move this in the array of selectors ?

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**